File: resource-2/ruby/bucketSort.rb
def bucketSort(a, n, bucket, m)
	for j in 0 ... m
		bucket[j] = 0
	end
	for i in 0 ... n
		bucket[a[i]] += 1
	end
	i = 0
	for j in 0 ... m
		for k in 0 ... bucket[j]
			a[i] = j
			i += 1
		end
	end
end

File: resource-2/java/bucketSort.java
public class bucketSort {
	public static void BucketSort(int[] a, int m) {
		int[] buckets = new int[m];
		for(int j = 0; j < m; ++j) {
			buckets[j] = 0;
		}
		for(int i = 0; i < a.length; ++i) {
			++buckets[a[i]];
		}
		for(int i = 0, j = 0; j < m; ++j) {
			for(int k = buckets[j]; k > 0; --k) {
				a[i++] = j;
			}
		}
	}
}
